StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/SpacingRules/SA1008CSharp7UnitTests.cs

```diff
@@ -978,5 +978,65 @@ public void TestMethod(params (string name, string value)[] options)
 
             await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
         }
+
+        [Fact]
+        [WorkItem(3117, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3117")]
+        public async Task TestTupleParameterAttributesAsync()
+        {
+            var testCode = @"
+namespace TestNamespace
+{
+    class NotNullAttribute : System.Attribute { }
+    class CustomAttribute : System.Attribute { }
+
+    class TestClass
+    {
+        void TestMethod1([NotNull] (string, string) tuple) { }
+
+        void TestMethod2([NotNull, Custom] (string, string) tuple) { }
+
+        void TestMethod3([NotNull][Custom] (string, string) tuple) { }
+
+        void TestMethod4([NotNull](string, string) tuple) { }
+
+        void TestMethod5([NotNull, Custom](string, string) tuple) { }
+
+        void TestMethod6([NotNull][Custom](string, string) tuple) { }
+    }
+}
+";
+
+            var fixedCode = @"
+namespace TestNamespace
+{
+    class NotNullAttribute : System.Attribute { }
+    class CustomAttribute : System.Attribute { }
+
+    class TestClass
+    {
+        void TestMethod1([NotNull] (string, string) tuple) { }
+
+        void TestMethod2([NotNull, Custom] (string, string) tuple) { }
+
+        void TestMethod3([NotNull][Custom] (string, string) tuple) { }
+
+        void TestMethod4([NotNull] (string, string) tuple) { }
+
+        void TestMethod5([NotNull, Custom] (string, string) tuple) { }
+
+        void TestMethod6([NotNull][Custom] (string, string) tuple) { }
+    }
+}
+";
+
+            DiagnosticResult[] expectedDiagnostics =
+            {
+                Diagnostic(DescriptorPreceded).WithLocation(15, 35),
+                Diagnostic(DescriptorPreceded).WithLocation(17, 43),
+                Diagnostic(DescriptorPreceded).WithLocation(19, 43),
```

Comment:
If possible, can you do one of the following?

1. If the `WithLocation(int)` overload is available, use this syntax:

    ```suggestion
                    Diagnostic(DescriptorPreceded).WithLocation(0),
                    Diagnostic(DescriptorPreceded).WithLocation(1),
                    Diagnostic(DescriptorPreceded).WithLocation(2),
    ```

    Then you can mark the locations 0, 1, and 2 above by replacing the `(` where the diagnostic occurs with `{|#0:(|}` (and using 1 and 2 for the others).

2. If the `WithLocation(int)` overload is _not_ available (meaning we are still on an older version of the testing library, just replace the calls to `WithLocation` with calls to `WithSpan`. I believe this is the correct change:

    ```suggestion
                    Diagnostic(DescriptorPreceded).WithSpan(15, 35, 15, 36),
                    Diagnostic(DescriptorPreceded).WithSpan(17, 43, 17, 44),
                    Diagnostic(DescriptorPreceded).WithSpan(19, 43, 19, 44),
    ```

Reply:
That's a great suggestion, thanks! The new markup syntax is already available in the test projects. Just pushed a commit which fixes that.